common/util/hal_snoop.c

```diff
@@ -0,0 +1,82 @@
+#include <zephyr.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <sys/printk.h>
+#include <device.h>
+#include "cmsis_os.h"
+#include "hal_snoop.h"
+
+const struct device *snoop_dev;
+uint8_t *snoop_data;
+static uint8_t *snoop_read_buffer;
+int snoop_read_num = 0;
+
+K_THREAD_STACK_DEFINE( snoop_thread , SNOOP_STACK_SIZE );
+struct k_thread snoop_thread_handler;;
+k_tid_t snoop_tid;
+
+struct k_mutex snoop_mutex;
+
+void snoop_init(){
+  snoop_dev = device_get_binding(DT_LABEL(DT_NODELABEL(snoop)));
+  if (!snoop_dev) {
+    printk("No snoop device found\n");
+    return;
+  }
+  return;
+}
+
+void copy_snoop_read_buffer( uint8_t offset ,int size_num, uint8_t *buffer ){
+  if ( ! k_mutex_lock(&snoop_mutex, K_MSEC(1000)) ){
+    memcpy( &buffer[0] , &snoop_read_buffer[offset] , size_num - offset );
+    memcpy( &buffer[ size_num - offset ] , &snoop_read_buffer[0] , offset );
+  }else{
+    printk("copy snoop buffer lock fail\n");
+  }
+  if ( k_mutex_unlock(&snoop_mutex) ){
+    printk("copy snoop buffer unlock fail\n");
+  }
+}
+
+void snoop_read(){
+  int rc , status;
```

Comment:
Status doesn't look like it's used, can you remove it.

Reply:
OK, we will remove it and PR again.